airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_32_8_001__AirbyteConfigDatabaseDenormalization.java

```diff
@@ -0,0 +1,883 @@
+                                                           final ConfigWithMetadata<StandardSync> standardSyncWithMetadata) {
+    final Field<UUID> id = DSL.field("id", SQLDataType.UUID.nullable(false));
+    final Field<UUID> connectionId = DSL.field("connection_id", SQLDataType.UUID.nullable(false));
+    final Field<UUID> operationId = DSL.field("operation_id", SQLDataType.UUID.nullable(false));
+    final Field<OffsetDateTime> createdAt = DSL.field("created_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false));
+    final Field<OffsetDateTime> updatedAt = DSL.field("updated_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false));
+
+    if (createTable) {
+      ctx.createTableIfNotExists("connection_operation")
+          .columns(id,
+              connectionId,
+              operationId,
+              createdAt,
+              updatedAt)
+          .constraints(primaryKey(id, connectionId, operationId),
+              foreignKey(connectionId).references("connection", "id"),
+              foreignKey(operationId).references("operation", "id"))
+          .execute();
+      LOGGER.info("connection_operation table created");
+    }
+    final StandardSync standardSync = standardSyncWithMetadata.getConfig();
+    for (final UUID operationIdFromStandardSync : standardSync.getOperationIds()) {
+      ctx.insertInto(DSL.table("connection_operation"))
+          .set(id, UUID.randomUUID())
+          .set(connectionId, standardSync.getConnectionId())
+          .set(operationId, operationIdFromStandardSync)
+          .set(createdAt, OffsetDateTime.ofInstant(standardSyncWithMetadata.getCreatedAt(), ZoneOffset.UTC))
+          .set(updatedAt, OffsetDateTime.ofInstant(standardSyncWithMetadata.getUpdatedAt(), ZoneOffset.UTC))
+          .execute();
+    }
+    LOGGER.info("connection_operation table populated with " + standardSync.getOperationIds().size() + " records");
+  }
+
+  private static <T> List<ConfigWithMetadata<T>> listConfigsWithMetadata(final AirbyteConfig airbyteConfigType,
+                                                                         final Class<T> clazz,
+                                                                         final DSLContext ctx) {
+    final Field<String> configId = DSL.field("config_id", SQLDataType.VARCHAR(36).nullable(false));
+    final Field<String> configType = DSL.field("config_type", SQLDataType.VARCHAR(60).nullable(false));
+    final Field<OffsetDateTime> createdAt = DSL.field("created_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false));
+    final Field<OffsetDateTime> updatedAt = DSL.field("updated_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false));
+    final Field<JSONB> configBlob = DSL.field("config_blob", SQLDataType.JSONB.nullable(false));
+    final Result<Record> results = ctx.select(asterisk()).from(DSL.table("airbyte_configs")).where(configType.eq(airbyteConfigType.name())).fetch();
```

Comment:
worth noting that we are making an assumption here that each table we are pulling can fully fit in memory.

Reply:
do you think it might cause problems and needs change?

Reply:
that's a great point charles, it might if the database has a ton of connections, sources or destinations.

I'll be curious to test this. I think we should test how this performs with a database with ten thousands connections to be safe. I don't expect the other resources to be used with the same level of scale.